anyTV/jenkins-github-skip-pr-by-title-plugin
# ZULU-44 CI > Skip Build function

## Ticket references
- [ZULU-44 CI > Skip Build function](https://freedom.myjetbrains.com/youtrack/issue/ZULU-44)

## Summary of changes
Add files for the plugin.

## Checklist
- [x] added unittests

## Testing
### Test Objectives
* Build the plugin to verify that the code builds properly.
* Install and use the plugin on Jenkins to verify that it does filter out pull requests containing the skip keywords.

### Test Steps

Log in to http://10.10.15.70:8181
```
username: freedom-ci
password: asdf
```

1. Go to [the list of recognized PRs for the sample repository](http://10.10.15.70:8181/job/aloremia-freedom/job/useless-basic-calc/view/change-requests), verify that both PR-8 and PR-9 are recognized.
1. Go to the GitHub page of [PR-8](https://github.com/aloremia-freedom/useless-basic-calc/pull/8) and [PR-9](https://github.com/aloremia-freedom/useless-basic-calc/pull/9), and note that PR-9 has the keyword "[skip-CI]" on the PR title.
1. Follow prerequisites, build, and installation steps at [README.md](https://github.com/anyTV/jenkins-github-skip-pr-by-title-plugin/tree/feature/initial-files#prerequisites).
1. Go to [the configure page for the sample job](http://10.10.15.70:8181/job/aloremia-freedom/configure).
1. Under *Projects* > *GitHub Organization* > *Behaviors*, add the behavior "Pull request title filtering behaviour" and then **Save** the configuration. The CI will scan and build the repo.
1. Once finished, go to [the list of recognized PRs for the sample repository](http://10.10.15.70:8181/job/aloremia-freedom/job/useless-basic-calc/view/change-requests) and verify that only PR-8 is there.
1. Go back to [the configure page for the sample job](http://10.10.15.70:8181/job/aloremia-freedom/configure) and **DELETE** the "Pull request title filtering behaviour". Save the configuration.
1. Go to [installed plugin page](http://10.10.15.70:8181/pluginManager/installed) and **Uninstall** "GitHub Skip PR by Title Plugin".
1. Go [here](http://10.10.15.70:8181/safeRestart) and restart Jenkins.
